apps/editor.planx.uk/src/@planx/components/SetFee/utils.ts

```diff
@@ -54,6 +55,37 @@ export const handleSetFees: HandleSetFees = ({
     fees[`${CALCULATED_FN}.VAT`] = calculatedVAT;
     fees[payable] = fees[payable] + calculatedVAT;
     fees[payableVAT] = calculatedVAT;
+
+    // It's possible for discretionary services with a VAT-able base application fee
+    //   to apply exemptions or reductions which are then also VAT-able
+    const { reduction, reductionVAT, exemption, exemptionVAT } =
+      getCalculatedReductionOrExemption({
+        ...passport.data,
+        // Ensure any calculations so far overwrite original incoming passport
+        [CALCULATED_FN]: calculated,
+        [`${CALCULATED_FN}.VAT`]: calculatedVAT,
+        [PAY_FN]: fees[payable],
+        [`${PAY_FN}.VAT`]: fees[payableVAT],
+      });
+
+    // VAT-able exemptions are always 100%, so simply subtract all VAT
+    if (exemption !== 0 && exemptionVAT !== 0) {
+      fees[payable] = fees[payable] - calculatedVAT;
+      fees[payableVAT] = fees[payableVAT] - calculatedVAT;
+    }
+
+    // VAT-able reductions can be any percent of calculated VAT
+    if (reduction !== 0 && reductionVAT !== 0) {
+      if (reduction === -calculated && reductionVAT === -calculatedVAT) {
+        // If 100% reduction, handle same as exemption
+        fees[payable] = fees[payable] - calculatedVAT;
+        fees[payableVAT] = fees[payableVAT] - calculatedVAT;
+      } else {
+        // Else only subtract percentage of all calculated VAT
+        fees[payable] = fees[payable] - (calculatedVAT + reductionVAT);
+        fees[payableVAT] = fees[payableVAT] - (calculatedVAT + reductionVAT);
+      }
+    }
```

Comment:
This is the most important change ! Now, if `SetFees` is toggled to "Apply VAT to calculated base application fee", it will additionally check for the presence of any VAT-able exemptions and reductions. 

"Payable" totals are then adjusted to account for any calculated VAT _and_ exemption or reduction VAT accordingly (planx-core has handled calculating the _amount_ of exemption or reduction VAT, now planx-new `SetFee` handles adjusting _total fees_). 

This happens in order _before_ any Fast Track, service charge, or payment processing fees are applied if applicable.